src/Invoice_GenUI/Invoice_GenUI/Models/Services/ClientService.cs

```diff
@@ -1,74 +1,58 @@
     {
         public async Task<List<ClientNameModel>> GetClientNames()
         {
-            using (var client = new HttpClient())
-            {
-                client.BaseAddress = new Uri("https://2023-invoice-gen.azurewebsites.net/");
-
-                var response = await client.GetAsync("clients");
-
-                response.EnsureSuccessStatusCode();
-
-                return await response.Content.ReadFromJsonAsync<List<ClientNameModel>>() ?? new();
-            }
+            return await SendHttpRequest<List<ClientNameModel>>("Clients") ?? new();
         }
         public async Task<List<CreateClientModel>> GetClientDetails()
         {
-            using (var client = new HttpClient())
-            {
-                client.BaseAddress = new Uri("https://2023-invoice-gen.azurewebsites.net/");
-
-                var response = await client.GetAsync("clients");
-
-                response.EnsureSuccessStatusCode();
-
-                return await response.Content.ReadFromJsonAsync<List<CreateClientModel>>() ?? new();
-            }
+            return await SendHttpRequest<List<CreateClientModel>>("Clients") ?? new();
         }
-        public async Task<CreateClientModel> GetSingleClientDetails(int index)
+        public async Task<CreateClientModel> GetSingleClientDetails(int id)
         {
-            using (var client = new HttpClient())
-            {
-                client.BaseAddress = new Uri("https://2023-invoice-gen.azurewebsites.net/");
-
-                var response = await client.GetAsync($"clients/{index}");
-
-                response.EnsureSuccessStatusCode();
-
-                return await response.Content.ReadFromJsonAsync<CreateClientModel>() ?? new();
-            }
+            return await SendHttpRequest<CreateClientModel>($"Clients/{id}") ?? new();
         }
         public async Task<bool> PutClient(CreateClientModel newClient)
         {
             bool result = false;
-            using (var client = new HttpClient())
-            {
-                client.BaseAddress = new Uri("https://2023-invoice-gen.azurewebsites.net/");
 
+            using (var client = CreateHttpClient())
+            {
                 var json = JsonSerializer.Serialize(newClient);
                 var content = new StringContent(json, Encoding.UTF8, "application/json");
 
+                var response = await client.PutAsync("Clients/Client", content);
 
-                var responseMessage = await client.PutAsync("Clients/Client", content);
-                if (responseMessage.IsSuccessStatusCode)
+                response.EnsureSuccessStatusCode();
+                if (response.IsSuccessStatusCode)
                 {
-                    var responseContent = await responseMessage.Content.ReadAsStringAsync();
                     result = true;
                 }
+            }
+            return result;
+        }
+        public async Task<bool> DeleteClient(int id)
```

Comment:
:question: if there is a base service which wraps the HttpClient, is it worth creating a generic version of this method too?